src/runtime_src/core/common/xclbin_parser.cpp

```diff
@@ -983,6 +1031,103 @@ get_vbnv(const axlf* top)
   return {vbnv, strnlen(vbnv, vbnv_length)};
 }
 
+std::string
+get_device_name(const char* xmlFile,size_t xmlFileSize)   
```

Comment:
The device name per XML is not currently extracted by xrt::xclbin, but please use same approach as used for project name (see common/api/xclbin_int.h, xrt_xclbin.cpp).